### PR TITLE
doc: HashiCorp Terraform Version Compatibility Matrix

### PR DESCRIPTION
**Pull request by Zuhairahmed**
## Description

Please include a summary of the fix/feature/change, including any relevant motivation and context.
Updates our external facing docs for HashiCorp Terraform Version Compatibility Matrix 

Link to any related issue(s): [CLOUDP-218730](https://jira.mongodb.org/browse/CLOUDP-218730) 

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue). Please, add the "bug" label to the PR.
- [ ] New feature (non-breaking change which adds functionality). Please, add the "enhancement" label to the PR.
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected). Please, add the "breaking change" label to the PR.
- [ ] This change requires a documentation update
- [x] Documentation fix/enhancement

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have read the [contribution guidelines](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/CONTRIBUTING.md)
- [x] I have checked that this change does not generate any credentials and that **they are NOT accidentally logged anywhere**.
- [ ] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirements
- [x] I have added any necessary documentation (if appropriate)
- [ ] I have run make fmt and formatted my code
- [ ] If changes include deprecations or removals, I defined an isolated PR with a relevant title as it will be used in the auto-generated changelog.
- [ ] If changes include removal or addition of 3rd party GitHub actions, I updated our internal document. Reach out to the APIx Integration slack channel to get access to the internal document.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -16,8 +16,21 @@ Bugs should be filed under the Issues section of this repo.
 
 Feature requests can be submitted at https://feedback.mongodb.com/forums/924145-atlas - just select the Terraform plugin as the category or vote for an already suggested feature.
 
-## Requirements
-- [Terraform Version](https://www.terraform.io/downloads.html) While we support minimum of HashiCorp Terraform version 1.0+ (as we have migrated to [Plugin Protocol Version 6](https://developer.hashicorp.com/terraform/plugin/terraform-plugin-protocol#protocol-version-6)), for your safety we strongly recommend only consuming versions of Terraform that are currently receiving Security/Maintenance Updates. For latest Terraform versions see: https://endoflife.date/terraform 
+## Requirements  
+- [HashiCorp Terraform Version](https://www.terraform.io/downloads.html) Compatibility Matrix
+  
+| HashiCorp Terraform Release | HashiCorp Terraform Release Date  | HashiCorp Terraform Full Support End Date  | MongoDB Atlas Support End Date |
```
Comment (lantoli) [inline README.md]: i would probably have the table only in one place and make reference to it, e.g. to: https://registry.terraform.io/providers/mongodb/mongodbatlas/latest/docs

File: website/docs/index.html.markdown [hunk 2]
```diff
@@ -179,13 +179,19 @@ In addition to [generic `provider` arguments](https://www.terraform.io/docs/conf
...
 For more information on configuring and managing programmatic API Keys see the [MongoDB Atlas Documentation](https://docs.atlas.mongodb.com/tutorial/manage-programmatic-access/index.html).
 
-## HashiCorp Terraform Version Requirement
-| MongoDB Atlas Provider version  | Required Terraform version |
-| ------------- | ------------- |
-| `1.12.0` and above  | `1.0+`  |
-| below `1.12.0`  | `0.13+` |
-
-* While we support minimum of HashiCorp Terraform version 1.0+ (as we have migrated to [Plugin Protocol Version 6](https://developer.hashicorp.com/terraform/plugin/terraform-plugin-protocol#protocol-version-6)), for your safety we strongly recommend only consuming versions of Terraform that are currently receiving Security/Maintenance Updates. For latest Terraform versions see: https://endoflife.date/terraform 
+## [HashiCorp Terraform Version](https://www.terraform.io/downloads.html) Compatibility Matrix
```
Comment (lantoli) [inline website/docs/index.html.markdown:182]: my feeling is that this info is very below in the page but i suppose it doesn't fit more above

**Review by lantoli — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (Zuhairahmed) [reply README.md]: makes sense, updated

File: website/docs/index.html.markdown (hunk 2 above)
Comment (Zuhairahmed) [reply website/docs/index.html.markdown:182]: played around with the spacing, hope this helps with readability

**Comment by lantoli** [comment]
LGTM

File: website/docs/index.html.markdown [hunk 3]
```diff
@@ -179,13 +179,20 @@ In addition to [generic `provider` arguments](https://www.terraform.io/docs/conf
...
+ 
+| HashiCorp Terraform Release | HashiCorp Terraform Release Date  | HashiCorp Terraform Full Support End Date  | MongoDB Atlas Support End Date |
+|:-------:|:------------:|:------------:|:------------:|
+| 1.7.x | 2024-01-17 | 2026-01-31 | 2026-01-31 |
+| 1.6.x | 2023-10-04 | 2025-10-31 | 2025-10-31 |
+| 1.5.x | 2023-06-12 | 2025-06-30 | 2025-06-30 |
+| 1.4.x | 2023-03-08 | 2025-03-31 | 2025-03-31 |
+| 1.3.x | 2022-09-21 | 2024-09-30 | 2024-09-30 |
+| 1.2.x | 2022-05-18 | 2024-05-31 | 2024-05-31 |
```
Comment (marcosuma) [inline website/docs/index.html.markdown:191]: would it make sense to add one more line to explicitly call out that <= 1.1.x is no longer supported?

File: website/docs/index.html.markdown [hunk 4]
```diff
@@ -179,13 +179,20 @@ In addition to [generic `provider` arguments](https://www.terraform.io/docs/conf
...
 For more information on configuring and managing programmatic API Keys see the [MongoDB Atlas Documentation](https://docs.atlas.mongodb.com/tutorial/manage-programmatic-access/index.html).
 
-## HashiCorp Terraform Version Requirement
-| MongoDB Atlas Provider version  | Required Terraform version |
-| ------------- | ------------- |
-| `1.12.0` and above  | `1.0+`  |
-| below `1.12.0`  | `0.13+` |
-
-* While we support minimum of HashiCorp Terraform version 1.0+ (as we have migrated to [Plugin Protocol Version 6](https://developer.hashicorp.com/terraform/plugin/terraform-plugin-protocol#protocol-version-6)), for your safety we strongly recommend only consuming versions of Terraform that are currently receiving Security/Maintenance Updates. For latest Terraform versions see: https://endoflife.date/terraform 
+## [HashiCorp Terraform Version](https://www.terraform.io/downloads.html) Compatibility Matrix
```
Comment (maastha) [inline website/docs/index.html.markdown:182]: [q] do we plan to include a mention in our next release guide about the Compatibility matrix to notify users? Or would the changelog suffice?

File: website/docs/index.html.markdown [hunk 5]
```diff
@@ -179,13 +179,20 @@ In addition to [generic `provider` arguments](https://www.terraform.io/docs/conf
...
+| 1.7.x | 2024-01-17 | 2026-01-31 | 2026-01-31 |
+| 1.6.x | 2023-10-04 | 2025-10-31 | 2025-10-31 |
+| 1.5.x | 2023-06-12 | 2025-06-30 | 2025-06-30 |
+| 1.4.x | 2023-03-08 | 2025-03-31 | 2025-03-31 |
+| 1.3.x | 2022-09-21 | 2024-09-30 | 2024-09-30 |
+| 1.2.x | 2022-05-18 | 2024-05-31 | 2024-05-31 |
+
+For the safety of our users, we require only consuming versions of HashiCorp Terraform that are currently receiving Security / Maintenance Updates. For more details see [Support Period and End-of-Life (EOL) Policy](https://support.hashicorp.com/hc/en-us/articles/360021185113-Support-Period-and-End-of-Life-EOL-Policy).   
+
+For latest HashiCorp Terraform versions see [here](https://endoflife.date/terraform ).
```
Comment (Zuhairahmed) [inline website/docs/index.html.markdown]:
```suggestion
HashiCorp Terraform versions that are not listed on this table are no longer supported by MongoDB Atlas. For latest HashiCorp Terraform versions see [here](https://endoflife.date/terraform ).
```

File: website/docs/index.html.markdown (hunk 3 above)
Comment (Zuhairahmed) [reply website/docs/index.html.markdown:191]: Added additional sentence below such that we no longer have mention specific version numbers as discussed

File: website/docs/guides/1.16.0-upgrade-guide.html.markdown [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+
+The Terraform MongoDB Atlas Provider version 1.16.0 has a number of new and exciting features.
+
+**New Resources, Data Sources, and Features:**
+
+
+
+**Deprecations and Removals:**
+
+- The HashiCorp Terraform Version Compatibility Matrix has been published. See full details at https://registry.terraform.io/providers/mongodb/mongodbatlas/latest/docs#hashicorp-terraform-version-compatibility-matrix
```
Comment (Zuhairahmed) [inline website/docs/guides/1.16.0-upgrade-guide.html.markdown]:
```suggestion
- The HashiCorp Terraform Version Compatibility Matrix has been published for the Terraform MongoDB Atlas Provider. See full details at https://registry.terraform.io/providers/mongodb/mongodbatlas/latest/docs#hashicorp-terraform-version-compatibility-matrix
```

File: website/docs/index.html.markdown (hunk 4 above)
Comment (Zuhairahmed) [reply website/docs/index.html.markdown:182]: Suggest we mention in both places with link to Terraform Docs. Added additional commit with updates to upcoming Upgrade Guide. ChangeLog will be pulled in automatically from the ChangeLog auto-generation, will also manually adjust as needed when we get there.

File: website/docs/index.html.markdown (hunk 5 above)
Comment (lantoli) [reply website/docs/index.html.markdown]: supported:   ... no longer supported by ...

**Review by AgustinBettati — APPROVED** [review]
LGTM

**Review by marcosuma — APPROVED** [review]

**Pull request merged**